File: C/toLinkedList.c
struct ListNode {
    int data;
    struct ListNode* next;
};

struct ListNode* toLinkedList(struct Queue* queue) {
    if (isEmpty(queue)) {
        return NULL;
    }
    struct ListNode* head = NULL;
    struct ListNode* tail = NULL;
    for (int i = queue->front; i <= queue->rear; i++) {
        struct ListNode* newNode = (struct ListNode*)malloc(sizeof(struct ListNode));
        newNode->data = queue->array[i];
        newNode->next = NULL;
        if (tail == NULL) {
            head = newNode;
            tail = newNode;
        } else {
            tail->next = newNode;
            tail = newNode;
        }
    }
    return head;
}

File: Java/toLinkedList.java
class ListNode {
    int data;
    ListNode next;

    public ListNode(int data) {
        this.data = data;
        this.next = null;
    }
}

public ListNode toLinkedList() {
    if (isEmpty()) {
        return null;
    }
    ListNode head = null;
    ListNode tail = null;
    for (int i = front; i <= rear; i++) {
        ListNode newNode = new ListNode(array.get(i));
        if (tail == null) {
            head = newNode;
            tail = newNode;
        } else {
            tail.next = newNode;
            tail = newNode;
        }
    }
    return head;
}
